cmd/kubernetes-discovery/pkg/cmd/server/start.go

```
@@ -135,9 +137,15 @@ func (o DiscoveryServerOptions) RunDiscoveryServer() error {
 		return err
 	}
 
+	kubeconfig, err := restclient.InClusterConfig()
+	if err != nil {
+		return err
+	}
+
 	config := apiserver.Config{
-		GenericConfig:     genericAPIServerConfig,
-		RESTOptionsGetter: &restOptionsFactory{storageConfig: &o.Etcd.StorageConfig},
+		GenericConfig:       genericAPIServerConfig,
+		RESTOptionsGetter:   &restOptionsFactory{storageConfig: &o.Etcd.StorageConfig},
+		CoreAPIServerClient: kubeclientset.NewForConfigOrDie(kubeconfig),
```

this should return an error